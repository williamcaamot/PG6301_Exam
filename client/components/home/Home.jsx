import React from "react";
function Home(){
    return<>
        <h2>welcome you must sign in to view chats</h2>
    </>
}

export default Home;