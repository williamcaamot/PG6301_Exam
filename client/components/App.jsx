import React, { createContext, useEffect, useState } from "react";
import { BrowserRouter, Route, Router, Routes } from "react-router-dom";
import Login from "./login/Login.jsx";
import Header from "./globals/Header.jsx";
import Home from "./home/Home.jsx";
import GoogleLoginCallback from "./login/GoogleLoginCallback.jsx";
import EntraIdLoginCallback from "./login/EntraIdLoginCallback.jsx";
import { Chat } from "./chat/Chat.jsx";
import PersonalProfilePage from "./profile/PersonalProfilePage.jsx";
import AddNewChatroom from "./chat/AddNewChatroom.jsx";
import Profile from "./profile/Profile.jsx";
import ExternalProfilePage from "./profile/ExternalProfilePage.jsx";
import editProfilePage from "./profile/EditProfilePage.jsx";
import EditProfilePage from "./profile/EditProfilePage.jsx";
import EditChatRoom from "./chat/EditChatRoom.jsx";
import ErrorMessage from "./globals/ErrorMessage.jsx";

export const AppContext = createContext();

export function App() {
  const [errorMessage, setErrorMessage] = useState();

  const [user, setUser] = useState(null);

  async function fetchUser() {
    try {
      const res = await fetch("/api/v1/login");
      if (res.status === 401) {
        setUser(undefined);
      } else {
        const { message, data } = await res.json();
        setUser(data);
      }
    } catch (e) {
      setErrorMessage(e.message);
    }
  }

  useEffect(() => {
    fetchUser();
  }, []);

  return (
    <AppContext.Provider value={{ user, setUser }}>
      <BrowserRouter>
        <ErrorMessage message={errorMessage} />
        <Header />
        <Routes>
          <Route
            path={"/login/callback/entraid"}
            element={<EntraIdLoginCallback />}
          />
          <Route
            path={"login/callback/google"}
            element={<GoogleLoginCallback />}
          />
          <Route path={"/login"} element={<Login />} />
          <Route path={"/profile"} element={<PersonalProfilePage />} />
          <Route
            path={"/profile/:email"}
            element={<ExternalProfilePage />}
          />{" "}
          <Route path={"/chat"} element={<Chat />} />
          <Route path={"/chatroom/add"} element={<AddNewChatroom />} />
          <Route path={"/chatroom/edit/:id"} element={<EditChatRoom />} />
          <Route path={"/profile/edit"} element={<EditProfilePage />} />
          <Route path={"/"} element={<Home />} />
        </Routes>
      </BrowserRouter>
    </AppContext.Provider>
  );
}
