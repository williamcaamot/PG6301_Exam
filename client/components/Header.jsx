import React, {useContext} from "react";
import {Link} from "react-router-dom";
import {AppContext} from "./App.jsx";

function Header() {

    const { user } = useContext(AppContext);

    return <>
        <div className={"flexWrapper"}>
            <div className={"headerWrapper"}>
                <div>
                    <h1>Banter Battle - Chat website</h1>
                </div>
                <div>
                    <nav>
                        <Link to="/" className={"mainMenuLink"}>Hjem</Link>
                        <Link to="/chat" className={"mainMenuLink"}>Chat</Link>
                        <Link to="/chatroom/add" className={"mainMenuLink"}>Add new chat room</Link>

                    </nav>
                </div>
                <div style={{flexDirection:"row-reverse"}}>
                    {user ?
                        <Link to="/profile" className={"mainMenuLink"}>Velkommen, {user.nickname}</Link>
                        :
                        <Link to="/login" className={"mainMenuLink"}>Logg inn</Link>
                    }

                </div>
            </div>
        </div>
    </>
}

export default Header;