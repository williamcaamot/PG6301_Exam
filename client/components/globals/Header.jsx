import React, {useContext} from "react";
import {Link} from "react-router-dom";
import {AppContext} from "../App.jsx";

function Header() {

    const { user } = useContext(AppContext);

    return <>
        <div className={"flexWrapper"}>
            <div className={"headerWrapper"}>
                <div style={{padding:"30px"}}>
                    <h1>ChatMania</h1>
                </div>
                <div>
                    <nav>
                        <Link to="/" className={"mainMenuLink"}>Hjem</Link>
                        <Link to="/chat" className={"mainMenuLink"}>Chat</Link>
                        <Link to="/chatroom/add" className={"mainMenuLink"}>Add new chat room</Link>
                        {user ?
                            <Link to="/profile" className={"mainMenuLink"}>Velkommen, {user.nickname}</Link>
                            :
                            <Link to="/login" className={"mainMenuLink"}>Logg inn</Link>
                        }
                    </nav>
                </div>
            </div>
        </div>
    </>
}

export default Header;