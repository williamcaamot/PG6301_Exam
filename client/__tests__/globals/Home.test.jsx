import renderer, { act } from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";
import Header from "../../components/globals/Header.jsx";
import { AppContext } from "../../components/App.jsx";
import Home from "../../components/home/Home.jsx";

describe("Home component", () => {
  it("Should display the home component", () => {
    const user = null;

    const component = renderer.create(
      <AppContext.Provider value={{ user }}>
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      </AppContext.Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
