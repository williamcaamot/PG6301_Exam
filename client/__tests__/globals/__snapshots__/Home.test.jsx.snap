// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should display the home component 1`] = `
<div
  className="pageContentWrapper"
>
  <div
    className="innerWrapper"
  >
    <div
      style={
        {
          "width": "100%",
        }
      }
    >
      <h2>
        Velkommen til denne chatte appen!
      </h2>
    </div>
    <p>
      Her kan alle registrerte brukere chatte med hverandre i forskjellige chat rom.
    </p>
    <p>
      Du kan logge inn med Google eller Microsoft Entra ID.
    </p>
  </div>
</div>
`;
